### PR TITLE
Fix multiple frontends with docker-compose --scale

### LINKED ISSUE #3145: Scaling with docker-compose --scale results in multiple frontends and backends
### Do you want to request a *feature* or report a *bug*?
As far as I can understand, this must be a bug.

### What did you do?
I created a simple `docker-compose.yaml`:

```yaml
version: '3'

services:
  scaler:
    image: registry:5000/static/master:latest

    restart: unless-stopped

    networks:
      - web

    labels:
      - "traefik.backend=foo"
      - "traefik.docker.network=web"
      - "traefik.frontend.rule=Host: scaler.test.ab.cd"
      - "traefik.enable=true"
      - "traefik.port=80"
      - "traefik.default.protocol=http"

networks:
  web:
    external: true
```

I then ran it with `docker-compose -d --scale scaler=3`.

### What did you expect to see?
I expected to see one frontend and one backend, the latter containing three docker instances.

### What did you see instead?

Træfik creates three separate frontends and three separate backends, and no load balancing takes place. The frontends are named `frontend-scaletest-scaler-1-foo-default`, `frontend-scaletest-scaler-2-foo-default` and so on, and the backends are named `backend-scaletest-scaler-1-foo-default` and so on.

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.5.2
Codename:     cancoillotte
Go version:   go1.9.4
Built:        2018-02-12_10:56:31AM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
debug = false

logLevel = "ERROR"
defaultEntryPoints = ["https","http"]

[entryPoints]
  [entryPoints.http]
  address = ":80"
    [entryPoints.http.redirect]
    entryPoint = "https"
  [entryPoints.https]
  address = ":443"
  [entryPoints.https.tls]
[retry]

[docker]
endpoint = "unix:///var/run/docker.sock"
domain = "test.kxweb.no"
watch = true
exposedbydefault = false

[acme]
email = "ola.nordmann@ab.cd"
storage = "/etc/traefik/acme/acme.json"
entryPoint = "https"
OnHostRule = true
[acme.httpChallenge]
entryPoint = "http"

[web]
address = ":8080"

# [accessLog]
```

### If applicable, please paste the log output at DEBUG level (`--logLevel=DEBUG` switch)

```
(paste your output here)
```

### PR DESCRIPTION
**Pull request by jbdoumenjou**
### What does this PR do?

This PR update the docker provider to fix an issue with the docker compose scale.

### Motivation

Fix #3145 

### More

- [x] Added/updated tests

### CODE REVIEW

File: integration/docker_compose_test.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+)
+
+// Docker test suites
+type DockerComposeSuite struct {
+	BaseSuite
+	project *docker.Project
+}
+
+func (s *DockerComposeSuite) SetUpSuite(c *check.C) {
+	s.createComposeProject(c, "minimal")
```
Comment (ldez) [inline integration/docker_compose_test.go]:
```go
s.createComposeProject(c, composeProject)
```

File: integration/docker_compose_test.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+func (s *DockerComposeSuite) TearDownSuite(c *check.C) {
+	// shutdown and delete compose project
+	if s.composeProject != nil {
+		s.composeProject.Stop(c)
+	}
+}
+
+func (s *DockerComposeSuite) TestComposeScale(c *check.C) {
+	const serviceCount = 2
+	const composeService = "whoami1"
```
Comment (ldez) [inline integration/docker_compose_test.go]: use `var` instead `const`

File: integration/docker_compose_test.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	body, err := ioutil.ReadAll(resp.Body)
+	c.Assert(err, checker.IsNil)
+
+	var provider types.Configuration
+	c.Assert(json.Unmarshal(body, &provider), checker.IsNil)
+
+	// check that we have only one backend with n servers
+	c.Assert(provider.Backends, checker.HasLen, 1)
+	myBackend := provider.Backends["backend-"+composeService+"-integrationtest"+composeProject]
+	c.Assert(myBackend.Servers, checker.HasLen, serviceCount)
```
Comment (ldez) [inline integration/docker_compose_test.go]: you need to check if the backend exists before check if the server exists in the backend.

File: provider/docker/config.go [hunk 4]
```diff
@@ -104,6 +105,18 @@ func (p *Provider) buildConfigurationV2(containersInspected []dockerData) *types
...
+func getServiceNameKey(container dockerData, swarmMode bool, segmentName string) string {
+	serviceNameKey := container.ServiceName
+
+	if values, err := label.GetStringMultipleStrict(container.Labels, labelDockerComposeProject, labelDockerComposeService); !swarmMode && err == nil {
+		serviceNameKey = values[labelDockerComposeService] + values[labelDockerComposeProject]
+	}
+
+	serviceNameKey += segmentName
+
+	return serviceNameKey
```
Comment (ldez) [inline provider/docker/config.go]:
```go
return serviceNameKey + segmentName
```

File: integration/docker_compose_test.go [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+func (s *DockerComposeSuite) TearDownSuite(c *check.C) {
+	// shutdown and delete compose project
+	if s.composeProject != nil {
+		s.composeProject.Stop(c)
+	}
+}
+
+func (s *DockerComposeSuite) TestComposeScale(c *check.C) {
+	var serviceCount = 2
```
Comment (ldez) [inline integration/docker_compose_test.go]: not needed, only one occurrence.

Comment (jbdoumenjou) [reply integration/docker_compose_test.go]: serviceCount is used [L44](https://github.com/containous/traefik/pull/3190/files/0c9ca7ec5462f6c9b98e9e4dfdba7d37dce762a6#diff-42a4390e47bc8cbbb2db8a2ce35202b5R44) and [L76](https://github.com/containous/traefik/pull/3190/files/31858a49d221493f5963b6a6f7a6f33f9430f06f#diff-42a4390e47bc8cbbb2db8a2ce35202b5R76)

**Pull request merged**